pandas/core/construction.py

```diff
@@ -773,6 +773,13 @@ def _try_cast(
             #  special case.
             return maybe_cast_to_datetime(arr, dtype)
             # TODO: copy?
+        from pandas import IntervalDtype
+
+        if (
+            isinstance(dtype, IntervalDtype)
+            and np.isnan(np.array(arr).astype("float")).any()
```

Comment:
can this check just be `isna(arr).any()`?

Reply:
Yes can